frontend/model/state.js

```diff
@@ -522,6 +544,18 @@ const getters = {
       return getters.ourUnreadMessages[chatRoomId]?.mentions || []
     }
   },
+  directMessageIDFromUsername (state, getters) {
+    return (username: string) => getters.mailboxContract.users[username]?.contractID
+  },
+  usernameFromDirectMessageID (state, getters) {
+    return (chatRoomId: string) => {
+      if (!getters.isDirectMessage(chatRoomId)) {
+        return
+      }
+      return Object.keys(getters.mailboxContract.users)
+        .find(username => getters.directMessageIDFromUsername(username) === chatRoomId)
+    }
```

Comment:
Couldn't there be multiple users in a DM?

Reply:
Adding guests is not implemented yet. And direct message is a channel between only two users.
If we invite guests, we should create a new chatroom (not DM) whose `type` is `CHATROOM_TYPES.INDIVIDUAL` and `privacyLevel` is `CHATROOM_PRIVACY_LEVEL.GROUP` in mailbox contract.

For a DM, there is only one partner.